setup.py

```diff
@@ -139,10 +132,77 @@ def _collect_requirements(requires):
 }
 
 extras_require["all"] = _collect_requirements(extras_require)
+extras_require["test"] = [
```

Comment:
These are more or less the same as `kedro-datasets[all]`, but since we're removing datasets from framework soon, I just copy pasted `test_requirements.txt` here and didn't aim to substantially change what we're doing. Most of these can go away in 0.19.0.